File: src/main/java/com/machfour/macros/names/ColumnUnits.java
package com.machfour.macros.names;

import com.machfour.macros.core.Column;
import com.machfour.macros.objects.NutritionData;
import com.machfour.macros.objects.Unit;

import org.jetbrains.annotations.NotNull;

import java.util.Collection;

public interface ColumnUnits {
    @NotNull
    Unit getUnit(@NotNull Column<NutritionData, Double> col);

    @NotNull
    Collection<Column<NutritionData, Double>> availableColumns();
}

File: src/main/kotlin/com/machfour/macros/names/ColumnUnits.kt
package com.machfour.macros.names

import com.machfour.macros.core.Column
import com.machfour.macros.objects.NutritionData
import com.machfour.macros.objects.Unit

interface ColumnUnits {
    fun getUnit(col: Column<NutritionData, Double>): Unit
    val columnsWithUnits: Collection<Column<NutritionData, Double>>
}